### ISSUE TITLE
[Feature request] Add spline editing tool

### ISSUE THREAD

**Issue by AttackButton** [turanszkij/WickedEngine#912]
One of the tools with the most possible uses, from creating roads for cars, curving the meshes of pipes and trees, to creating paths for moving platforms to travel and much more.

So, if it is possible to implement such a tool in the editor, would it be possible to add these buttons below to more easily select the control points of the curve, just like Unreal Engine does? Probably only the center three buttons would be enough: the center one selecting all points, and the other ones jumping to the next or previous point of the curve. Additionally, the ability to attach metadata to this entity with the points of a curve would be beneficial too.

![image](https://github.com/user-attachments/assets/060bc07e-1dc7-46e3-9574-7c876bad27ac)

**Comment by turanszkij** [comment]
Yes, this would be a great addition, I also want this but not sure when I get to work on it.

**Comment by cryham** [comment]
Exactly, this is the one feature I miss. Would be great to have.
It was something I did implement early in [Stunt Rally](https://github.com/stuntrally/stuntrally3), for roads, and their bridges. It evolved into road pipes. Later for rivers and waterfalls and lastly also for walls, fences etc.
It has 4 LODs and segment merging by length. Later I started with grid merging, likely better, just worse for editor.
I think this all could be done better now in modern engine, maybe even with tessellation.

**Comment by turanszkij** [comment]
Work in progress: https://github.com/turanszkij/WickedEngine/pull/1080

**Comment by turanszkij** [comment]
Completed.
